inc/cxplat.hpp

```diff
@@ -0,0 +1,87 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    C++ header only cxplat wrappers
+
+--*/
+
+#ifndef CXPLATCPP_H
+#define CXPLATCPP_H
+
+#include "cxplat.h"
+#include "cxplat_sal_stub.h"
+
+typedef void* CxPlatCallback(
+    _Inout_ void* Context
+);
+
+class CxPlatAsync {
+private:
+    struct CxPlatAsyncContext {
+        void* UserContext;
+        CxPlatCallback *UserCallback;
+        void* ReturnValue;
+    };
+
+    static
+    CXPLAT_THREAD_CALLBACK(CxPlatAsyncWrapperCallback, Context)
+    {
+        struct CxPlatAsyncContext *AsyncContext = (struct CxPlatAsyncContext*)Context;
+        AsyncContext->ReturnValue = AsyncContext->UserCallback(AsyncContext->UserContext);
+        CXPLAT_THREAD_RETURN(0);
+    }
+
+    CXPLAT_THREAD Thread {0};
+    CXPLAT_THREAD_CONFIG ThreadConfig {0};
+    struct CxPlatAsyncContext AsyncContext {0};
+    bool Initialized = false;
+    bool ThreadCompleted = false;
+public:
+    CxPlatAsync(CxPlatCallback Callback, void* UserContext = nullptr) noexcept {
+        AsyncContext.UserContext = UserContext;
+        AsyncContext.UserCallback = Callback;
+        AsyncContext.ReturnValue = nullptr;
```

Comment:
Should be unnecessary, right?

Reply:
They are necessary to hide the thread callback signature difference between all platforms.